sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/StringExpressionsSuite.scala

```diff
@@ -1832,16 +1833,22 @@ class StringExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
   }
 
   test("SPARK-33468: ParseUrl in ANSI mode should fail if input string is not a valid url") {
```

Comment:
This test duplicates the check in `UrlFunctionsSuite`, actually. Let's move it to `UrlFunctionsSuite`, and use `parse_url()` in ANSI/non-ANSI modes.

Reply:
You mean add test case in UrlFunctionsSuite to use sql("SELECT parse_url(...)") command with ANSI disabled? There is already one test case in UrlFunctionSuite with ANSI disabled, using df.selectExpr("parse_url()") method.

Reply:
I care mostly of the case when the ANSI mode is enabled. Let's deduplicate the code for the case, and check the error in the end-to-end case only. The goal is to cover the error class by a test.

> There is already one test case in UrlFunctionSuite with ANSI disabled, using df.selectExpr("parse_url()") method.

ok. If it checks an invalid url.